build.sbt

```diff
@@ -1,120 +1,110 @@
-  "org.scorexfoundation"         %% "scrypto"                       % "2.1.7",
-  "org.scorexfoundation"         %% "scorex-util"                   % "0.1.6",
-  "com.muquit.libsodiumjna"      % "libsodium-jna"                  % "1.0.4" exclude ("org.slf4j", "slf4j-log4j12") exclude ("org.slf4j", "slf4j-api"),
-  "net.java.dev.jna"             % "jna"                            % "5.5.0", //see https://github.com/muquit/libsodium-jna/#update-your-projects-pomxml
-  "com.github.alanverbner"       %% "bip39"                         % "0.1",
-  "com.rklaehn"                  %% "radixtree"                     % "0.5.1",
-  "com.nrinaudo"                 %% "kantan.csv-generic"            % kantanCsvVersion,
-  "com.nrinaudo"                 %% "kantan.csv-java8"              % kantanCsvVersion,
-  "com.typesafe.akka"            %% "akka-testkit"                  % akkaVersion % Test exclude ("com.typesafe", "config"),
-  "com.typesafe.akka"            %% "akka-http-testkit"             % akkaHttpVersion % Test exclude ("com.typesafe", "config"),
-  "org.scalatest"                %% "scalatest"                     % "3.0.5" % "it, test",
-  "com.stephenn"                 %% "scalatest-json-jsonassert"     % "0.0.3" % "it, test",
-  "org.scalamock"                %% "scalamock"                     % "4.1.0" % "it, test",
-  "org.testcontainers"           % "postgresql"                     % "1.12.3" % "it, test",
-  "com.softwaremill.diffx"       %% "diffx-scalatest"               % "0.3.3" % "it, test",
-  "org.http4s"                   %% "http4s-blaze-client"           % http4sVersion % IntegrationTest,
-  "org.http4s"                   %% "http4s-dsl"                    % http4sVersion % IntegrationTest,
-  "org.http4s"                   %% "http4s-circe"                  % http4sVersion % IntegrationTest,
-  compilerPlugin("com.github.ghik" % "silencer-plugin" % silencerVersion cross CrossVersion.full),
-  "com.github.ghik" % "silencer-lib" % silencerVersion % Provided cross CrossVersion.full
-)
+lazy val conseil = (project in file("."))
+  .disableAssembly()
+  .aggregate(common, api, lorre, lorreSchema, smokeTests)
 
-excludeDependencies ++= Seq(
-  "org.consensusresearch" %% "scrypto"
-)
+lazy val common = (project in file("conseil-common"))
+  .settings(
+    name := "conseil-common",
+    libraryDependencies ++= Dependencies.conseilCommonInclude,
+    excludeDependencies ++= Dependencies.conseilCommonExclude,
+    coverageExcludedPackages := Seq(
+          "<empty>",
+          "tech.cryptonomic.conseil.common.io.*",
+          "tech.cryptonomic.conseil.common.tezos.Tables",
+          "tech.cryptonomic.conseil.common.config.Security"
+        ).mkString(";")
+  )
+  .settings(Defaults.itSettings)
+  .settings(
+    scalacOptions += "-P:silencer:pathFilters=common/src/main/scala/tech/cryptonomic/conseil/common/tezos/Tables.scala"
+  )
+  .configs(IntegrationTest)
+  .enableBuildInfo()
+  .disableAssembly()
```

Comment:
Previously, since there was only one project, entire project has been assembled so there was nothing to disable. I think that by default this plugin is disabled indeed, so I just removed this method and it works as expected.

Reply:
Ah, but you are asking about `disableAssembly()` at `conseil`, you may actually be right there. Let me investigate this part.

Reply:
I did investigated that, the conclusion is that, the plugin automatically enables assembly for all projects. Turning it off, at root level gives nothing. I managed to found the issue on GitHub for this problem [here](https://github.com/sbt/sbt-assembly/issues/175). There is no solution only workaround to disable plugin for specific modules.

I see different behavior on assembly. With the setup we have (but without disabling it at root, I'll remove it), we are only building modules we want. Otherwise, we are testing and assembling everything, which is not what we want.

Reply:
good to know, thanks.